### PR TITLE
Feature rest application configuration endpoint

### PR DESCRIPTION
**Pull request by RetGal**
Implements #373

### CODE REVIEW

File: AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/dtos/ConfigurationDTO.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+
+@Data
+@XmlRootElement(name = "configuration")
+@XmlAccessorType(XmlAccessType.FIELD)
+@JsonAutoDetect()
+public class ConfigurationDTO {
```
Comment (yvespp) [inline AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/dtos/ConfigurationDTO.java]: @RetGal is a separte dto realy needed, can't you annotation and return ApplicationConfigurationInfo, ConfigurationKeyValuePair and ConfigKey directly?
To obfuscate you could add a method to the existing service (like getObfuscated), there should already be a method that does this in the JSF module.

File: AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/settings/SettingsRest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+public class SettingsRest {
+
+    final static String OBFUSCATED = "***************";
+
+    @Inject
+    ApplicationVersionService applicationVersionService;
+
+    @GET
+    @ApiOperation(value = "Get Liima configuration information")
+    public List<ConfigurationDTO> getAppInfo() {
```
Comment (yvespp) [inline AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/settings/SettingsRest.java]: Better name it getAppConfig so it corresponds to the buissines method?

File: AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/dtos/ConfigurationDTO.java (hunk 1 above)
Comment (RetGal) [reply AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/dtos/ConfigurationDTO.java]: I'm trying to figure out to get the properies of ConfigKey correctly into JSON/XML.
Obfuscation is done in the JSF-view directly, not in a methos:
`<h:outputText value="***************" rendered="${configItem.key.secretValue}" />`

**Comment by yvespp** [comment]
Somehow I'm getting a exception on the deployment screen:
```
Caused by: com.fasterxml.jackson.databind.JsonMappingException: failed to lazily initialize a collection of role: ch.puzzle.itc.mobiliar.business.releasing.entity.ReleaseEntity.resources, could not initialize proxy - no Session (through reference chain: java.util.ArrayList[0]->ch.mobi.itc.mobiliar.rest.dtos.ResourceGroupDTO["releases"]->java.util.ArrayList[0]->ch.puzzle.itc.mobiliar.business.releasing.entity.ReleaseEntity["resources"])
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:378)
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:338)
        at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:342)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:686)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:157)
        at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContents(IndexedListSerializer.java:119)
        at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:79)
        at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:18)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:672)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:678)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:157)
        at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serializeContents(CollectionSerializer.java:149)
        at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:112)
        at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:25)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:253)
        at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize(ObjectWriter.java:1425)
        at com.fasterxml.jackson.databind.ObjectWriter.writeValue(ObjectWriter.java:930)
        at org.jboss.resteasy.plugins.providers.jackson.ResteasyJackson2Provider.writeTo(ResteasyJackson2Provider.java:207)
        at org.jboss.resteasy.core.interception.AbstractWriterInterceptorContext.writeTo(AbstractWriterInterceptorContext.java:131)
        at org.jboss.resteasy.core.interception.ServerWriterInterceptorContext.writeTo(ServerWriterInterceptorContext.java:60)
        at org.jboss.resteasy.core.interception.AbstractWriterInterceptorContext.proceed(AbstractWriterInterceptorContext.java:120)
        at org.jboss.resteasy.security.doseta.DigitalSigningInterceptor.aroundWriteTo(DigitalSigningInterceptor.java:145)
        at org.jboss.resteasy.core.interception.AbstractWriterInterceptorContext.proceed(AbstractWriterInterceptorContext.java:124)
        at org.jboss.resteasy.plugins.interceptors.encoding.GZIPEncodingInterceptor.aroundWriteTo(GZIPEncodingInterceptor.java:100)
        at org.jboss.resteasy.core.interception.AbstractWriterInterceptorContext.proceed(AbstractWriterInterceptorContext.java:124)
        at org.jboss.resteasy.core.ServerResponseWriter.writeNomapResponse(ServerResponseWriter.java:98)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeResponse(SynchronousDispatcher.java:473)
        ... 47 more
Caused by: org.hibernate.LazyInitializationException: failed to lazily initialize a collection of role: ch.puzzle.itc.mobiliar.business.releasing.entity.ReleaseEntity.resources, could not initialize proxy - no Session
        at org.hibernate.collection.internal.AbstractPersistentCollection.throwLazyInitializationException(AbstractPersistentCollection.java:587)
        at org.hibernate.collection.internal.AbstractPersistentCollection.withTemporarySessionIfNeeded(AbstractPersistentCollection.java:204)
        at org.hibernate.collection.internal.AbstractPersistentCollection.readSize(AbstractPersistentCollection.java:148)
        at org.hibernate.collection.internal.PersistentSet.size(PersistentSet.java:143)
        at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:102)
        at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:25)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:672)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:678)
        ... 70 more
```
Calling `curl -X GET "http://localhost:8080/AMW_rest/resources/resources" -H "accept: application/json"` gives the same result.
Tested with the image liimaorg/liima:feature_rest_application_configuration_endpoint
It seems that it trys to serialize the resources from a release, but the should be excluded...

**Comment by RetGal** [comment]
The issue was only reproducible with the docker image. The addition of explicit @JsonIgnoreProperties and removal of @JsonAutoDetect fixed the issue for this environment.

**Pull request merged**